examples/ramberg_osgood.py

```diff
@@ -0,0 +1,666 @@
+
+    def evaluate(
+        self,
+        del_t: float,
+        del_strain: np.ndarray,
+        stress: np.ndarray,
+        tangent: np.ndarray,
+        history: np.ndarray,
+    ) -> None:
+
+        stress_view = stress.reshape(-1, self.stress_dim)
+        tangent_view = tangent.reshape(-1, self.stress_dim ** 2)
+
+        for n, eps in enumerate(del_strain.reshape(-1, self.stress_dim)):
+            # eps = strain at time t + delta t
+            tr_eps = np.sum(eps[:3])
+            eps_dev = eps - tr_eps * self.I2 / 3
+            ev = np.sqrt(2.0 / 3.0 * np.dot(eps_dev, eps_dev))
+
+            if ev < self.zero_strain_tolerance:
+                # return elastic tangent
+                stress_view[n] = self.Cel @ eps
+                tangent_view[n] = self.Cel.flatten()
+            else:
+                # compute correct tangent and stress
+                # stress at time t
+                sig = stress_view[n]
+                tr_sig = np.sum(sig[:3])
+                sig_dev = sig - tr_sig * self.I2 / 3
+                # equivalent stress at time t is used as initial guess
+                sv_initial = np.sqrt(3.0 / 2.0 * np.dot(sig_dev, sig_dev))
+
+                # stress at time t + delta t
+                if sv_initial <= self.sigy:
+                    sv = sv_initial
+                else:
+                    # initial_guess is > sigy
+                    sv = (self.sigy ** (self.n - 1.0) * self.e * ev / self.alpha) ** (1.0 / self.n)
+
+                def f(x):
+                    stuff = 1.0 / (2.0 * self.mu) + 3.0 / (2.0 * self.e) * self.alpha * (x / self.sigy) ** (
+                        self.n - 1.0
+                    )
+                    return stuff * 2.0 / 3.0 * x - ev
+
+                def df(x):
+                    return 1.0 / (3.0 * self.mu) + self.n * self.alpha / self.e * (x / self.sigy) ** (self.n - 1.0)
+
+                s = f(sv)
+                ds = df(sv)
+
+                niter = 0
+                while abs(f(sv)) > self.sv_tol:
+                    sv = sv - s / ds
+                    s = f(sv)
+                    ds = df(sv)
+                    niter += 1
+                    if niter > self.maxiter:
+                        break
+
+                sig_dev = 2.0 * sv / 3.0 / ev * eps_dev
+                tr_sig = self.k * tr_eps
+                sig = tr_sig * self.I2 / 3.0 + sig_dev
+                stress_view[n] = sig
+
+                nenner = sv / (3.0 * self.mu) + self.alpha * self.n * self.sigy / self.e * ((sv / self.sigy) ** (self.n))
+                tangent = 2 * sv / 3 / ev * (
+                    self.I4
+                    - 2.0
+                    / 3.0
+                    / ev
+                    * (1.0 / ev - 1.0 / nenner)
+                    * np.outer(eps_dev, eps_dev)
+                ) + 1.0 / 3.0 * (self.k - 2 * sv / (3 * ev)) * np.outer(self.I2, self.I2)
+                tangent_view[n] = tangent.flatten()
+
+    def history_dim(self) -> int:
```

Comment:
Why did you not define the `stress_strain_dim`, `geometric_dim` and `constraint` methods/properties?